test/addons/argocd/start

```diff
@@ -0,0 +1,64 @@
+#!/usr/bin/env python3
+
+# SPDX-FileCopyrightText: The RamenDR authors
+# SPDX-License-Identifier: Apache-2.0
+
+import os
+import sys
+import json
+import time
+
+from drenv import kubectl
+from drenv import minikube
+from drenv import commands
+from drenv import cache
+
+
+def deploy(cluster):
+    print("Deploying argocd")
+    kubectl.apply("--filename", "argocd-ns.yaml", context=cluster)
+    path = cache.path("addons/argocd-install.yaml")
+    cache.fetch(".", path)
+    kubectl.apply("--filename", path, "--namespace", "argocd", context=cluster)
+
+
+def wait(cluster):
+    print("Waiting until all deployments are available")
+    kubectl.wait(
+        "deploy",
+        "--all",
+        "--for=condition=Available",
+        "--namespace=argocd",
+        "--timeout=300s",
+        context=cluster,
+    )
+
+
+if len(sys.argv) != 4:
+    print(f"Usage: {sys.argv[0]} hubcluster drcluster1 drcluster2")
+    sys.exit(1)
+
+os.chdir(os.path.dirname(__file__))
+cluster = sys.argv[1]
```

Comment:
Why do we deploy only on one managed cluster? Are you trying to support pull mode? Shouldn't we deploy on both managed clusters?

It seems that the commit message or maybe this file is missing some documentation explaining what we are trying to do and why.

In other addons the common way is to install on cluster which is the single argument passed to the addon. This works for most addons.

Some addons (like submariner, volsync, and rbd-mirror) need all or both managed clusters, and the addon install stuff on more than one cluster.

Reply:
After some research, by only installing argocd in the hub cluster, the hub cluster would push the application to the dr clusters. So current implementation is using the push mode. 

If want to use pull mode, we need install argocd on all clusters, and install additional pull controller as mentioned by doc : https://www.redhat.com/en/blog/introducing-the-argo-cd-application-pull-controller-for-open-cluster-management

Also the Application template inside the ApplicationSet needs add additional label and annotation to enable the pull mode.
```
labels:
	apps.open-cluster-management.io/pull-to-ocm-managed-cluster: 'true'
annotations:
	argocd.argoproj.io/skip-reconcile: 'true'
	apps.open-cluster-management.io/ocm-managed-cluster: '{{name}}'
```

As suggested by Shyam, we will use pull mode later.